### PR TITLE
bugzilla: add new functions to bugzilla client

### PR DESCRIPTION
**Pull request by AlexNPavel**
These new functions will be needed for a new bugzilla plugin
feature to clone a bug when a cherrypick of a PR is created.

/cc @stevekuznetsov

### CODE REVIEW

File: prow/bugzilla/types.go [hunk 1]
```diff
@@ -107,6 +109,81 @@ type Bug struct {
 	Whiteboard string `json:"whiteboard,omitempty"`
 }
 
+// BugCreate holds the info needed to create a new bug
```
Comment (stevekuznetsov) [inline prow/bugzilla/types.go:112]: I know I didn't put every possible JSON field for a GetBug call into the Bug struct -- but we don't need any of the ones I missed for create? Nice.

File: prow/bugzilla/client.go [hunk 2]
```diff
@@ -158,6 +158,103 @@ func (c *client) UpdateBug(id int, update BugUpdate) error {
...
+	} else {
+		// This cases _shouldn't_ happen. But just in case...
+		newBug.Description = fmt.Sprintf("This is a clone of Bug #%d.", bug.ID)
+		// We don't know whether this bug should be private. Default to private in case this bug contains private information.
+		newBug.CommentIsPrivate = true
+	}
+	return newBug
+}
+
+func (c *client) CloneBug(bugID int) (int, error) {
```
Comment (stevekuznetsov) [inline prow/bugzilla/client.go:222]: I'm surprised we need this -- and not

```go
func (c *client) CloneBug(bug Bug) (int, error) {
```

File: prow/bugzilla/types.go [hunk 3]
```diff
@@ -107,6 +109,81 @@ type Bug struct {
...
+	// Platform is the name of the platform (hardware) that the bug was filed against.
+	Platform string `json:"platform,omitempty"`
+	// Priority is the priority of the bug.
+	Priority string `json:"priority,omitempty"`
+	// Product is the name of the product this bug is in.
+	Product string `json:"product,omitempty"`
+	// QAContact is the login name of the current QA Contact on the bug.
+	QAContact string `json:"qa_contact,omitempty"`
+	// Resolution is the current resolution of the bug, or an empty string if the bug is open.
+	Resolution string `json:"resolution,omitempty"`
```
[the remark this answers is not in the document]
Comment (AlexNPavel) [reply prow/bugzilla/types.go:147]: Technically yes. That would happen if we cloned a bug for instance.

[the remark this answers is not in the document]
Comment (petr-muller) [reply prow/bugzilla/types.go:147]: :+1:

**Comment by AlexNPavel** [comment]
I've removed the `CloneBug` function from the client. That will instead be moved to the bugzilla plugin

**Pull request merged**